spec/models/miq_server/worker_management/kubernetes_spec.rb

```diff
@@ -270,6 +273,52 @@
               expect(server.worker_manager.sync_starting_workers).to include(worker)
             end
           end
+
+          context "with a worker that doesn't have a system_uid yet" do
+            let(:system_uid) { nil }
+
+            before { server.worker_manager.sync_config }
+
+            context "without a pod" do
+              let(:current_pods) { {} }
+
+              it "returns the worker as still starting" do
+                expect(server.worker_manager.sync_starting_workers).to include(worker)
+              end
+
+              context "with a worker that has been starting longer than the starting_timeout" do
+                let(:last_heartbeat) { 20.minutes.ago.utc }
+
+                it "marks the worker as not responding" do
+                  # Make sure that #find_worker returns our instance of worker that
+                  # that stubs the #stop_container method.
+                  expect(server.worker_manager).to receive(:find_worker).with(worker).and_return(worker)
```

Comment:
NOTE it looks a little weird to have `.with(worker).and_return(worker)` here but that is just the way the matcher works, without this line the worker object passed in to `stop_worker` does not have the mock `expect(worker).to receive(:stop_container)`  applied and it actually drops into ContainerOrchestrator to try to delete the container.  This ensures that `stop_worker` has the right object with the stubbed methods.